lib/ui/src/core/layout.js

```diff
@@ -69,7 +130,54 @@ export default function({ store }) {
         };
       });
     },
+
+    setOptions: options => {
+      const { layout, ui, selectedPanel } = store.getState();
+
+      if (options) {
+        const updatedLayout = {
+          ...layout,
+          ...pick(options, Object.keys(layout)),
+          ...checkDeprecatedLayoutOptions(options),
+        };
+
+        const updatedUi = {
+          ...ui,
+          ...pick(options, Object.keys(ui)),
+          ...checkDeprecatedThemeOptions(options),
+        };
+
+        store.setState(
+          {
+            layout: updatedLayout,
+            ui: updatedUi,
+            selectedPanel: options.panel || options.selectedPanel || selectedPanel,
+          },
+          { persistence: 'permanent' }
+        );
```

Comment:
Does this mean we perma-persist all options, including the ones that were session-persisted above?

I'm OK with this for now for simplicity, but we might want to do something more subtle in the future (which will probably require improvements to the persistence API)

Reply:
Yeah I briefly experimented with persisting all the layout & ui options, but decided to not increase the size of this PR even more.

But it was actually really nice, things like full-screen, nav, panelPosition being preserved over refreshes...

On my todo list after V5 launch.